test/e2e/azure_timesync.go

```diff
@@ -94,3 +101,149 @@ func AzureTimeSyncSpec(ctx context.Context, inputGetter func() AzureTimeSyncSpec
 		return kinderrors.AggregateConcurrent(testFuncs)
 	}, thirty, thirty).Should(Succeed())
 }
+
+const (
+	nsenterWorkloadFile = "workloads/nsenter/daemonset.yaml"
+)
+
+// AzureDaemonsetTimeSyncSpec implements a test that verifies time synchronization is healthy for
+// the nodes in a cluster. It uses a privileged daemonset and nsenter instead of SSH.
+func AzureDaemonsetTimeSyncSpec(ctx context.Context, inputGetter func() AzureTimeSyncSpecInput) {
```

Comment:
Should this just replace the SSH implementation? what's the upside of keeping the SSH validation around for other clusters?

Reply:
I didn't want to be too presumptuous, but I'd say yes. I don't know what we'd do for Windows but I assume chrony doesn't apply there anyway.

I was hoping to follow up and make this an easy way to do other on-host validation, since you get access to basically anything you want.

Reply:
I also kind of wanted to get some validation from merging this first and making sure they were both reliable, but it seems to work well already to me. Too used to debugging flakes in the other e2e 😄

Reply:
> I was hoping to follow up and make this an easy way to do other on-host validation, since you get access to basically anything you want.

could you please file an issue so we can have context for having the two side by side in the meantime in case you don't get to it right away?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1543